lib/curlable.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const child_process = require('child_process');
+const kill = require('tree-kill');
+
+const readStreamByLines = (stream, callback) => {
+  var chunk = '';
+  stream.on('data', (data) => {
+    chunk += data.toString();
+    var i;
+    while ((i = chunk.indexOf('\n')) != -1) {
+      callback(chunk.substr(0, i));
+      chunk = chunk.substr(i + 1);
+    }
+  });
+};
+
+const generateRandomQueryId = () => {
+  var s = '';
+  for (var i = 0; i < 8; ++i) {
+    s += String.fromCharCode(65 + Math.floor(Math.random() * 26)); // A..Z.
+    if (i == 3) {
+      s += '-';
+    }
+  }
+  return s;
+};
+
+const curlable = function(cmdline, inputLogger) {
+  this.logger = inputLogger || (() => {});
+  this.spawnedProcess = child_process.spawn('/bin/bash', ['-c', cmdline]);
+
+  this.queriesQueue = null;
+
+  this.nextLineHandler = null;
+
+  const self = this;
+  readStreamByLines(this.spawnedProcess.stdout, (line) => {
+    if (!self.nextLineHandler) {
+      console.error('FATAL: No `nextLineHandler`, the tool' +
+        ' is generating output when it is not expected.');
+      console.error('ERROR: ' + line);
+      process.exit(-1);
```

Comment:
This output and exit is unexpected from within a class that has callbacks API (logger etc.); I'd expect a callback to be called with an error object.

Reply:
C'mon, it's just a simple tool, v0.0.1.